File: TravelPlannerApp/TripService/Models/Trip.cs
using System.Diagnostics;

namespace TripService.Models
{
    public class Trip
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal Budget { get; set; }
        public string Notes { get; set; }
        public int UserId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Destination> Destinations { get; set; }
        public ICollection<Activity> Activities { get; set; }
        public ICollection<Expense> Expenses { get; set; }
        public ICollection<ChecklistItem> ChecklistItems { get; set; }
    }
}

File: frontend/src/models/Trip.js
export class Trip {
  constructor(id, name, description, startDate, endDate, budget, notes, userId, createdAt, totalExpenses, remainingBudget) {
    this.id = id
    this.name = name
    this.description = description
    this.startDate = startDate
    this.endDate = endDate
    this.budget = budget
    this.notes = notes
    this.userId = userId
    this.createdAt = createdAt
    this.totalExpenses = totalExpenses
    this.remainingBudget = remainingBudget
  }

}